llvm/test/CodeGen/X86/is_fpclass.ll

```diff
@@ -240,18 +240,22 @@ entry:
 define i1 @isfinite_f(float %x) {
 ; X86-LABEL: isfinite_f:
 ; X86:       # %bb.0: # %entry
-; X86-NEXT:    movl $2147483647, %eax # imm = 0x7FFFFFFF
-; X86-NEXT:    andl {{[0-9]+}}(%esp), %eax
-; X86-NEXT:    cmpl $2139095040, %eax # imm = 0x7F800000
-; X86-NEXT:    setl %al
+; X86-NEXT:    flds {{[0-9]+}}(%esp)
```

Comment:
It looks like for targets like X86 this behavior is bad. As it requires materialization of infinity value, it produces worse code.

Is there an example when this patch is profitable?

Can we confine this behavior to the profitable cases?

Reply:
The x64 case below is fewer instructions. I can't bring myself to care about x86_32 enough to reason about the regressions